code/modules/mob/living/simple_animal/hostile/headcrustation.dm

```diff
@@ -0,0 +1,35 @@
+/mob/living/simple_animal/hostile/headcrustation
+	name = "Headcrab"
+	desc = "It looks confused, and somewhat lost."
+	icon_state = "crabby"
+	icon_living = "crabby"
+	icon_dead = "crabby_rip"
+	gender = NEUTER
+	health = 25
+	maxHealth = 25
+	melee_damage_lower = 0
+	melee_damage_upper = 0
+	attack_verb_continuous = "chomps"
+	attack_verb_simple = "chomp"
+	attack_sound = 'sound/weapons/bite.ogg'
+	faction = list("creature")
+	robust_searching = 1
+	stat_attack = DEAD
+	obj_damage = 0
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	speak_emote = list("squeaks")
+	ventcrawler = VENTCRAWLER_ALWAYS
+	ranged = TRUE
+	ranged_message = "Leaps"
+	projectiletype = /obj/projectile/crab
+
+/mob/living/simple_animal/hostile/headcrustation/Shoot(atom/targeted_atom)
+	. = ..()
+	if(istype(., /obj/projectile/crab))
+		var/obj/projectile/crab/creb = .
+		creb.takenHealth = bruteloss
+	qdel(src)
+
+/mob/living/simple_animal/hostile/headcrustation/crowbar_act(mob/living/user, obj/item/I)
+	. = ..()
+	adjustHealth(10) //You get it
```

Comment:
This might not behave as you expect it to.
It will drain 10 integrity without an animation or anything, and then proceed to attack, causing the potential attack and damage of a dead mob. If it gibs/deletes that will cause issues.

Reply:
How would I go about resolving that?

Reply:
Well, you could The way other objects like `/obj/structure/spacevine` seems to be doing it is by redefining `attacked_by()` but this is old code, and not a great standard.
The ideal way would be to refactor `attacked_by()` to allow for special weaknesses when handling damage.

Reply:
Ahhh, I'll look into it. May put this on hold, I'm not sure yet

Reply:
Take your time, poke if you need help.